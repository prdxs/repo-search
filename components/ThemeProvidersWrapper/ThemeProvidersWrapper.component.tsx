import React from 'react';
import { ThemeProvider } from 'styled-components';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core';
import CssBaseline from '@material-ui/core/CssBaseline';

import theme from '@/theme';
import { IComponentProps } from '@/typings/common';

const ThemeProvidersWrapper: React.FC<IComponentProps> = ({ children }) => (
  <ThemeProvider theme={theme}>
    <MuiThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </MuiThemeProvider>
  </ThemeProvider>
);

export default ThemeProvidersWrapper;
