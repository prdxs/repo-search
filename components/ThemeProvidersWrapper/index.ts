export { default } from './ThemeProvidersWrapper.component';
