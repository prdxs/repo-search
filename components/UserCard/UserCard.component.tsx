import React from 'react';
import clsx from 'clsx';
import styled, { css } from 'styled-components';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import Avatar from '@material-ui/core/Avatar';

import { IUserCardProps } from './UserCard.typings';

const UserCard: React.FC<IUserCardProps> = ({ className, style, avatar }) => {
  return (
    <Card className={clsx('UserCard-root', className)} style={style}>
      <CardHeader
        avatar={
          <Avatar
            aria-label="avatar"
            className="UserCard-avatar"
            data-testid="UserCard-avatar"
            src={avatar}
          >
            N/A
          </Avatar>
        }
      />
    </Card>
  );
};

const StyledSearchBar = styled(UserCard)`
  &.UserCard-root {
    .UserCard-avatar {
      ${({ theme }) => {
        const size = theme.spacing(14);

        return css`
          height: ${size};
          width: ${size};
        `;
      }}
    }
  }
`;

export default StyledSearchBar;
