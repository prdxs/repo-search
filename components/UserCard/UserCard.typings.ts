import { IComponentProps } from '@/typings/common';

export interface IUserCardProps extends IComponentProps {
  avatar?: string;
  name?: string;
  bio?: string;
  location?: string;
}
