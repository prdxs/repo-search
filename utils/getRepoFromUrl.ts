const getRepoFromUrl = (url: string): { owner: string; repo: string } => {
  const parts = url.split('/');
  const len = parts.length;
  const owner = parts[len - 2];
  const repo = parts[len - 1];

  return { owner, repo };
};

export default getRepoFromUrl;
