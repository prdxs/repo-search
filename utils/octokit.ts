import { Octokit } from '@octokit/rest';

const octokit = new Octokit({
  baseUrl: 'https://api.github.com',
  auth: process.env.NEXT_PUBLIC_GH_TOKEN,
});

export default octokit;
