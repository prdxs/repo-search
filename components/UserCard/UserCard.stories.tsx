import React from 'react';
import { withKnobs, select, text, number } from '@storybook/addon-knobs';

import StorybookWrapper from '@/components/StorybookWrapper';
import UserCard from './UserCard.component';

export default {
  title: 'UserCard',
  component: UserCard,
  decorators: [withKnobs],
};

export const Default: React.FC = () => {
  const avatar = select(
    'avatar',
    {
      None: undefined,
      Ruben:
        'https://avatars0.githubusercontent.com/u/16763209?s=460&u=3e17b9f9b36b8b486b31cbf67181070cd9593212&v=4',
    },
    undefined
  );

  return (
    <StorybookWrapper>
      <UserCard
        style={{ width: 380 }}
        avatar={avatar}
        name={text('name', '')}
        bio={text('bio', 'I am super coool!')}
        location={text('location', 'Manchester')}
        repoCount={number('repoCount', 0)}
      />
    </StorybookWrapper>
  );
};
