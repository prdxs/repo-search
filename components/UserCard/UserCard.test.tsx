import React from 'react';
import { mount } from 'enzyme';

import { render } from '@/test/utils';
import UserCard from '@/components/UserCard';
import ThemeProvidersWrapper from '@/components/ThemeProvidersWrapper';

describe('UserCard', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<UserCard />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the not available letters if the avatar is not provided', () => {
    const { queryByText } = render(<UserCard />);
    const avatar = queryByText('N/A');

    expect(avatar).toBeInTheDocument();
  });

  it('should render the avatar if provided', () => {
    const avatarUrl =
      'https://avatars0.githubusercontent.com/u/16763209?s=460&u=3e17b9f9b36b8b486b31cbf67181070cd9593212&v=4';
    const wrapper = mount(
      <ThemeProvidersWrapper>
        <UserCard avatar={avatarUrl} />{' '}
      </ThemeProvidersWrapper>
    );

    expect(wrapper.find('img').prop('src')).toEqual(avatarUrl);
  });

  it('should render the name of the user if provided', () => {
    const name = 'Donald Trump';
    const { queryByText } = render(<UserCard name={name} />);

    const nameEl = queryByText(name);

    expect(nameEl).toBeInTheDocument();
  });

  it("should render 'Unknown' if the name of the user is not provided", () => {
    const { queryByText } = render(<UserCard />);

    const nameEl = queryByText('Unknown');

    expect(nameEl).toBeInTheDocument();
  });

  it('should render the bio if provided', () => {
    const { queryByText } = render(<UserCard bio="Cool bio" />);

    const nameEl = queryByText(/Cool bio/);

    expect(nameEl).toBeInTheDocument();
  });

  it('should render the location if provided', () => {
    const { queryByText } = render(<UserCard location="Manchester" />);

    const nameEl = queryByText(/Manchester/);

    expect(nameEl).toBeInTheDocument();
  });

  it('should render repoCount if provided', () => {
    const { queryByText } = render(<UserCard repoCount={10} />);

    const nameEl = queryByText(/10/);

    expect(nameEl).toBeInTheDocument();
  });
});
