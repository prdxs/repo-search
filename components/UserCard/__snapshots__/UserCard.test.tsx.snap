// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    style="padding: 2rem;"
  >
    <div
      class="MuiPaper-root MuiCard-root UserCard-root UserCardcomponent__StyledSearchBar-sc-1aw9vkf-0 kpDwyL MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-avatar"
        >
          <div
            aria-label="avatar"
            class="MuiAvatar-root MuiAvatar-circle UserCard-avatar MuiAvatar-colorDefault"
            data-testid="UserCard-avatar"
          >
            N/A
          </div>
        </div>
        <div
          class="MuiCardHeader-content"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Unknown 
          </h6>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
